### PR TITLE
Azure OpenAI: minimal, partial specification for Whisper transcription/translation

### PR DESCRIPTION
**Pull request by trrwilson**

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/models/audio.tsp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+@doc("""
+The configuration information for an audio transcription request.
+""")
+@added(ServiceApiVersions.v2023_09_01_Preview)
+model AudioTranscriptionOptions {
+  @doc("""
+  The audio data to transcribe. This must be the binary content of a file in one of the supported media formats:
+  flac, mp3, mp4, mpeg, mpga, m4a, ogg, wav, webm.
+  """)
+  @projectedName("csharp", "AudioData")
```
Comment (mssfang) [inline specification/cognitiveservices/OpenAI.Inference/models/audio.tsp:42]: Is possible all languages use the same name,  `AudioData`? 
If not, .NET and Java can align the name,
```
@projectedName("java", "audioData")```

File: specification/cognitiveservices/OpenAI.Inference/models/audio.tsp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  """)
+  temperature?: float32;
+
+  @doc("""
+  The model to use for this transcription request.
+  """)
+  // Implementation note: developer-facing specification of deployment or model by clients should be controlled either
+  // via an operation parameter or by this request body field -- but only one of those. This field should be hidden by
+  // clients if operation parameters are used and populated into the request body on an as-needed basis.
+  @projectedName("csharp", "InternalNonAzureModelName")
```
Comment (mssfang) [inline specification/cognitiveservices/OpenAI.Inference/models/audio.tsp:77]: Question: Is possible we can do
```
@projectedName("json", "model")
internalNonAzureModelName?: string;
```
So all languages could algin with the same name.

File: specification/cognitiveservices/OpenAI.Inference/models/audio.tsp [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  flac, mp3, mp4, mpeg, mpga, m4a, ogg, wav, webm.
+  """)
+  @projectedName("csharp", "AudioData")
+  file: bytes;
+
+  @doc("""
+  The requested format of the transcription response data, which will influence the content and detail of the result.
+  """)
+  @projectedName("json", "response_format")
+  responseFormat?: AudioTranscriptionFormat;
```
Comment (jpalvarezl) [inline specification/cognitiveservices/OpenAI.Inference/models/audio.tsp:49]: According to the swagger I believe this should default to `"json"`

`responseFormat?: AudioTranscriptionFormat = AudioTranscriptionFormat.json;`

Comment (trrwilson) [reply specification/cognitiveservices/OpenAI.Inference/models/audio.tsp:49]: Thanks for raising this! Per IM conversation, it looks like we have inconsistent representation of service-side default values across the board -- a few things that are documented service-side defaults *do* encode that in TypeSpec (and the downstream generated OpenAPI v2 documents), but many more *don't*.

I filed this work item to track the holistic pass, as one or another it seems like our current treatment must be wrong:
https://github.com/Azure/azure-rest-api-specs/issues/25820

I'll leave this one as-is for now if it's not causing a problem, as we'll need to clean things up in any eventuality and "no churn" helps in the immediate term.

File: specification/cognitiveservices/OpenAI.Inference/models/audio.tsp (hunk 1 above)
Comment (trrwilson) [reply specification/cognitiveservices/OpenAI.Inference/models/audio.tsp:42]: We should consult with @johanste and others on how we'd like to approach naming. I recall Johan expressing a desire for the TypeSpec names to match the wire format as closely as possible (ideally, never needing/using `@projectedName("json", "...")` -- with *possible* minor exceptions to handle things like camel vs. snake-case deviations (which is the balance this file aims for at the moment).

I'm all for standardizing one way or another, but we should ensure we standardize the correct way (and consistently). In the interim, language-specific `@projectedName` alignments seem like the right way to go.

**Review by jpalvarezl — APPROVED** [review]

**Pull request merged**